$OpenBSD: patch-src_winwidget_c,v 1.3 2011/05/04 11:43:31 dcoppa Exp $

Use NETWM for fullscreen windows only
(commit bce956523f56b88ec04c074cd55041c5b9a81c90)

--- src/winwidget.c.orig	Wed May  4 13:20:06 2011
+++ src/winwidget.c	Wed May  4 13:20:06 2011
@@ -774,15 +774,17 @@ void winwidget_resize(winwidget winwid, int w, int h)
 			winwid->w = (w > scr->width) ? scr->width : w;
 			winwid->h = (h > scr->height) ? scr->height : h;
 		}
-		/* XResizeWindow(disp, winwid->win, winwid->w, winwid->h); */
-		XGetWindowAttributes(disp, winwid->win, &attributes);
-		XTranslateCoordinates(disp, winwid->win, attributes.root,
-				      -attributes.border_width -
-				      attributes.x,
-				      -attributes.border_width - attributes.y, &tc_x, &tc_y, &ignored_window);
-		winwid->x = tc_x;
-		winwid->y = tc_y;
-		XMoveResizeWindow(disp, winwid->win, tc_x, tc_y, winwid->w, winwid->h);
+		if (winwid->full_screen) {
+			XGetWindowAttributes(disp, winwid->win, &attributes);
+			XTranslateCoordinates(disp, winwid->win, attributes.root,
+						-attributes.border_width -
+						attributes.x,
+						-attributes.border_width - attributes.y, &tc_x, &tc_y, &ignored_window);
+			winwid->x = tc_x;
+			winwid->y = tc_y;
+			XMoveResizeWindow(disp, winwid->win, tc_x, tc_y, winwid->w, winwid->h);
+		} else
+			XResizeWindow(disp, winwid->win, winwid->w, winwid->h);
 
 		winwid->had_resize = 1;
 		XFlush(disp);
